### ISSUE TITLE
EVO Paramètres Controller

### ISSUE THREAD

**Issue by rbcreation** [wakdev/slash-cms#45]
SUGGESTION - sl_controller et sla_controller :
- Remplacer :
  function _construct(&$core_class_ref,$module_id,$module_name,$module_params=null) {
  $this->slash = $core_class_ref;
  $this->module_id = $module_id;
  $this->module_name = $module_name;
  $this->setParams($module_params);
  $this->sla_construct();
  }
- Par :
  function _construct(&$core_class_ref,$module,$config) {
  $this->slash = $core_class_ref;
  $this->module = $module;
  $this->config = $config;
  $this->setParams($module["params"]);
  $this->sla_construct();
  }

Dans le core, dans le initialize_modules(), on balance le $row plutot que le $row["id"], $row["name"] etc... C'est plus souple si jamais on veux récupérer d'autres infos du module. Il faut aussi balancer le $config contenant les infos du site. Ca va servir pour plus tard.
Par contre, ca fait changer tous les appels comme par exemple $this->controller->module_name en $this->controller->module["name"] dans les modules. Sauf si on le redéfinie dans le __construct() du controller.

**Comment by wakdev** [comment]
Je suis d'accord sur le row passé lors de la construction du module.
Mais la config doit rester dans le noyau pour des raisons de sécurité, pour permettre de bloquer la récupération de certaines valeurs.

Exemple : un module qui pourrait afficher en clair le login / mdp de la base de donnée...

**Comment by wakdev** [comment]
D'ailleurs j'ai rien bloqué pour l'instant. Je dois le faire ><

**Comment by wakdev** [comment]
Réglé ^^ : https://github.com/wakdev/slash-cms/commit/df18c57c58d8c7379cac058b15ee2dd0e6c65d65

**Comment by rbcreation** [comment]
Attention, je ne parle pas d'envoyer la config correspondant au fichier sl_config.php avec les infos concernant la base de données. Surtout pas.
Je parlais d'envoyer l'attribut $this->config, correspondant à la table sl_config dans la base. C'est pour récupérer par exemple le nom du template par défaut. $this->config["template_name"].

**Comment by wakdev** [comment]
Oui ca peut être pratique, mais je pense que le noyau doit garder le contrôle de ce genre de paramètres.
Sur cette logique tu peux altérer certains paramètres lorsque certains modules les sollicite. 
Ex : refuser l'accès à un paramètre pour un module en particulier ^^

Non ?

**Comment by rbcreation** [comment]
Yes, c'est vrai qu'on peut récupérer cela en passant directement par le core référent : $this->slash->config["template_name"].
